src/Dao/FacultyDaoImpl.java

```diff
@@ -0,0 +1,271 @@
+package com.masai.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+
+import com.masai.bean.Faculty;
+import com.masai.custom.AutoPassGen;
+import com.masai.custom.ConsoleColors;
+import com.masai.exceptions.FacultyException;
+import com.masai.utility.DBconn;
+
+public class FacultyDaoImpl implements FacultyDao {
+
+	
+	// Add New Faculty into Database
+	@Override
+	public String addFaculty(Faculty faculty) throws FacultyException{
+
+		String message = ConsoleColors.RED+"Data Not Inserted..."+ConsoleColors.RESET;
+		
+		try(Connection conn = DBconn.provideConnection()){
+			
+			Statement statement = conn.createStatement();
+            String sql = "SELECT MAX( facultyId ) FROM faculty";
+            
+            ResultSet result = statement.executeQuery(sql);
+            
+            Integer id = 0;
+            
+            if(result.next()) {
+            	id = result.getInt("max( facultyId )");
+            }
+            
+            id = id + 1;
+            String text = String.format("%03d", id);
+			String fname = faculty.getFname().toLowerCase();
+			String username = fname + text;
+			String password = AutoPassGen.genPass(8);
+			
+			PreparedStatement ps1 = conn .prepareStatement("insert into Faculty(facultyFname, facultyLname, " +
+					"facultyAddress, facultyState, facultyPin, mobile, email, username, password) values(?,?,?,?,?,?,?,?,?)");
+			
+			ps1.setString(1, faculty.getFname());
```

Comment:
lof of shortcut naming conventions. please provide meaningful namings. your code should easily be able to understand by other engineers